transaction/std/src/transaction_builder.rs

```diff
@@ -57,6 +57,21 @@ pub struct TxOutContext {
     pub shared_secret: RistrettoPublic,
 }
 
+/// Signing data for external library
```

Comment:
```suggestion
/// A structure containing an unsigned transaction, together with the data required to sign it that does
/// not involve the spend private key.
/// The idea is that this can be generated without having the spend private key, and then transferred
/// to an offline/hardware service that does have the spend private key, which can then be used together
/// with the data here to produce a valid, signed Tx. 
```

Reply:
So I guess this is just an UnsignedTx, it may or may not have all the data required depending on what is in the `InputRing`s

Reply:
I think the input rings no longer really contain the one-time private key, they only contain the `OnetimeKeyDeriveData` now? So they might, if it's for the gift code path i guess. But maybe it's okay not to mention the edge case

Reply:
well, it's an unsigned transaction that may or may not be signable on its own, depending on what is in the InputRings. I'll see if I can update the comment to reflect that